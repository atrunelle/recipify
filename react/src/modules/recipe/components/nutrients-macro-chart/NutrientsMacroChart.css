.nutrients-macro-chart {
  width: 100%;
  display: flex;
  flex-direction: column; }
  @media (min-width: 600px) {
    .nutrients-macro-chart {
      flex-direction: row; } }
