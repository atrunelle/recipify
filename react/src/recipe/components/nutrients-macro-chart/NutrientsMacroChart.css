.graph {
  width: 100%;
  display: flex;
  flex-direction: column; }
  @media (min-width: 600px) {
    .graph {
      flex-direction: row; } }
  .graph__bar {
    font-weight: 700;
    text-align: center;
    padding: 20px;
    color: #fff; }
    .graph__bar--fat {
      background: #FF5722; }
    .graph__bar--protein {
      background: #FFC107; }
    .graph__bar--carbs {
      background: #3F51B5; }
