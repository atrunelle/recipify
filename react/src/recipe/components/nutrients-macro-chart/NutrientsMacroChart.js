import React, { Component } from 'react';

class NutrientsMacroChart extends Component {
  render() {
    return (
      <div>Macro chart</div>
    );
  }
}

export default NutrientsMacroChart;