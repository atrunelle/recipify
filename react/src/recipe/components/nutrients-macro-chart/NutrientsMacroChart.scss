@import "breakpoint-sass/stylesheets/breakpoint";
@import "../../../assets/scss/global/site-settings";

.nutrients-macro-chart {
  width: 100%;
  display: flex;

  flex-direction: column;

  @include breakpoint($medium) {
    flex-direction: row;    
  }
}