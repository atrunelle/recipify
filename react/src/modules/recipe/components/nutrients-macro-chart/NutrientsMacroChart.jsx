import React, { Component } from 'react';
import './NutrientsMacroChart.css';
import BarGraph from '../../../../components/bar-graph/BarGraph';

class NutrientsMacroChart extends Component {

  render() {
    if (!this.props.nutritionData) return false;
    const bars = this.props.nutritionData.map((data, index) => {
      const percentage = data.totalQuantity.percentage;
      const label = data.totalQuantity.label;

      return (
        <BarGraph
          key={index}
          label={label}
          percentage={percentage}/>
      )
    });
    return (
      <div className="nutrients-macro-chart">
        {bars}
      </div>
    );
  }
}

export default NutrientsMacroChart;