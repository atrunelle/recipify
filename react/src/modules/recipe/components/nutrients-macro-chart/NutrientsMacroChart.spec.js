import React from 'react';
import { shallow } from 'enzyme';
import NutrientsMacroChart from './NutrientsMacroChart';

describe('Component: NutrientsMacroChart', () => {
  it('should not render if nutritionData prop is not passed', () => {
    const component = shallow(<NutrientsMacroChart />);
    expect(component.html()).toBeNull();
  });

  it('should match snapshot', () => {
    const mockNutritionData = [{
      totalQuantity: {
        label: 'Fat',
        quantity: 10,
        percentage: 12,
        unit: 'gr',
      },
    }];

    const component = shallow(
      <NutrientsMacroChart nutritionData={mockNutritionData}/>
    );
    expect(component.html()).toMatchSnapshot();
  });
});