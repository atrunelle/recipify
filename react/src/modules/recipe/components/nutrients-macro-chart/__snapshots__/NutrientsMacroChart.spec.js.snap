// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: NutrientsMacroChart should match snapshot 1`] = `"<div class=\\"nutrients-macro-chart\\"><div style=\\"width:12%\\" class=\\"bar-graph bar-graph--fat\\"><p>Fat 12%</p></div></div>"`;
